src/ar/ar.json

```diff
@@ -22,25 +18,2079 @@
+      "mapping": {
+        "base": "ؤ"
+      }
+    },
+    "ؤّ": {
+      "mapping": {
+        "base": "ؤ"
+      }
+    },
+    "ؤَّ": {
+      "mapping": {
+        "base": "ؤ"
+      }
+    },
+    "ؤِّ": {
+      "mapping": {
+        "base": "ؤ"
+      }
+    },
+    "ؤُّ": {
+      "mapping": {
+        "base": "ؤ"
+      }
+    },
+    
     "ؤ": {
       "mapping": {
         "base": "و"
       }
     },
-    "إ": {
+    
+    "يَ": {
       "mapping": {
-        "base": "ا"
+        "base": "ي"
+      }
+    },
+    "يِ": {
+      "mapping": {
+        "base": "ي"
+      }
+    },
+    "يُ": {
+      "mapping": {
+        "base": "ي"
+      }
+    },
+    "يْ": {
+      "mapping": {
+        "base": "ي"
+      }
+    },
+    "يً": {
+      "mapping": {
+        "base": "ي"
+      }
+    },
+    "يٍ": {
+      "mapping": {
+        "base": "ي"
+      }
+    },
+    "يٌ": {
+      "mapping": {
+        "base": "ي"
+      }
+    },
+    "يّ": {
+      "mapping": {
+        "base": "ي"
+      }
+    },
+    "يَّ": {
+      "mapping": {
+        "base": "ي"
+      }
+    },
+    "يِّ": {
+      "mapping": {
+        "base": "ي"
+      }
+    },
+    "يُّ": {
+      "mapping": {
+        "base": "ي"
+      }
+    },
+    
+    "ئَ": {
+      "mapping": {
+        "base": "ئ"
+      }
+    },
+    "ئِ": {
+      "mapping": {
+        "base": "ئ"
+      }
+    },
+    "ئُ": {
+      "mapping": {
+        "base": "ئ"
+      }
+    },
+    "ئْ": {
+      "mapping": {
+        "base": "ئ"
+      }
+    },
+    "ئً": {
+      "mapping": {
+        "base": "ئ"
+      }
+    },
+    "ئٍ": {
+      "mapping": {
+        "base": "ئ"
+      }
+    },
+    "ئٌ": {
+      "mapping": {
+        "base": "ئ"
       }
     },
+    "ئّ": {
+      "mapping": {
+        "base": "ئ"
+      }
+    },
+    "ئَّ": {
+      "mapping": {
+        "base": "ئ"
+      }
+    },
+    "ئِّ": {
+      "mapping": {
+        "base": "ئ"
+      }
+    },
+    "ئُّ": {
+      "mapping": {
+        "base": "ئ"
+      }
+    },
+    
     "ئ": {
       "mapping": {
         "base": "ي"
       }
     },
-    "آ": {
+    
+    "ىَ": {
       "mapping": {
-        "base": "ا"
+        "base": "ى"
+      }
+    },
+    "ىِ": {
+      "mapping": {
+        "base": "ى"
+      }
+    },
+    "ىُ": {
+      "mapping": {
+        "base": "ى"
+      }
+    },
+    "ىْ": {
+      "mapping": {
+        "base": "ى"
+      }
+    },
+    "ىً": {
+      "mapping": {
+        "base": "ى"
+      }
+    },
+    "ىٍ": {
+      "mapping": {
+        "base": "ى"
+      }
+    },
+    "ىٌ": {
+      "mapping": {
+        "base": "ى"
+      }
+    },
+    "ىّ": {
+      "mapping": {
+        "base": "ى"
+      }
+    },
+    "ىَّ": {
+      "mapping": {
+        "base": "ى"
+      }
+    },
+    "ىِّ": {
+      "mapping": {
+        "base": "ى"
+      }
+    },
+    "ىُّ": {
+      "mapping": {
+        "base": "ى"
+      }
+    },
+    
+    "ى": {
+      "mapping": {
+        "base": "ي"
       }
```

Comment:
This entry seems off to me, I don't know the language, so please correct me if I'm wrong; but, this looks like a base character being mapped to a base + diacritic.

Reply:
ي is a whole letter form, not seperatable into the shape above and the two dots below.

if you were to place the cursor before اَ and hit backspace, the top diacritic slash mark would be removed, leaving the base letter ا

this is because اَ is a combination of ا (the base) and َ  the diacritic.


if you were to place the cursor before ي and hit backspace, nothing would be left, the character would be deleted as it is a whole letter inseperable into a base letter and diacritical mark.

ئ and ى are whole letter form variations of the base ي
as آ أ إ ٱ are of the base ا

Reply:
Yes, but are the two dots a diacritic?

For example, an `ë` character is a combined form and named [LATIN SMALL LETTER E WITH DIAERESIS](http://www.fileformat.info/info/unicode/char/eb/index.htm). It has a unicode value of `\u00eb`. There is also a [COMBINING DIAERESIS](http://www.fileformat.info/info/unicode/char/0308/index.htm) which when combined with a base letter of `e` (`ë`; `e\u0308`) creates a symbol that exactly matches the `\u00eb` character, but when pressing backspace, you'll find there are two characters.

What I'm trying to say is that just because it's a single unicode character doesn't necessarily mean that the character doesn't include a diacritic.

For me, even after looking up [ARABIC LETTER YEH](http://www.fileformat.info/info/unicode/char/064a/index.htm), I'm still not sure if that is a base character or base including a diacritic... the comment on that page makes it an interesting case:

> loses its dots when used in combination with U+0654
retains its dots when used in combination with other combining marks

I'll leave this determination to you since I'm clueless. Thanks again for all your hard work.

Reply:
more examples of dotted diacrticless base characters:

ب ت ث ج خ ذ ز ض ظ ش ن

of which some outside the language might have mistaken the following to be diacrtical combinations of another base while they are not:
ج خ | ح
ذ | د
ظ | ط
ض | ص
ظ | ط
ش | س
ز | ر

Reply:
Ok, sounds great! Is this PR now ready to be merged?

Reply:
I say yes

Reply:
> if the ي were to include diacritics

Do those characters with diacritcs need to be included in the database then?